drivers/hmis_supplemental/app/models/hmis_supplemental/data_set.rb

```diff
@@ -0,0 +1,81 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HmisSupplemental
+  class DataSet < GrdaWarehouseBase
+    self.table_name = 'hmis_supplemental_data_sets'
+    has_paper_trail
+
+    belongs_to :remote_credential, class_name: 'GrdaWarehouse::RemoteCredentials::S3'
+    accepts_nested_attributes_for :remote_credential
+
+    belongs_to :data_source, class_name: 'GrdaWarehouse::DataSource'
+    validates :owner_type, inclusion: { in: ['enrollment', 'client'] }
+    has_many :field_values,
+             class_name: 'HmisSupplemental::FieldValue',
+             foreign_key: 'data_set_id',
+             dependent: :restrict_with_exception
+    validates :name, presence: true
+    validates :object_key, presence: true
+    has_many :group_viewable_entities,
+             class_name: 'GrdaWarehouse::GroupViewableEntity',
+             as: :entity,
+             dependent: :destroy
+
+    scope :viewable_by, ->(user) do
+      # not supporting legacy role base permissions
+      return none unless user.using_acls?
+
+      # first ask the WH db for all collections containing this resource type
+      gve_scope = GrdaWarehouse::GroupViewableEntity.where(entity_type: sti_name)
+      collection_ids = gve_scope.distinct.pluck(:collection_id)
+
+      # include global system collection
+      collection_ids += [Collection.system_collection(:data_sources)&.id].compact
```

Comment:
This doesn't make sense to me.  `DataSet`s are explicitly assigned in collections.  The "system" collection of **All Data Sets** should take care of this.  This would be granting anyone who has access to **All Data Sources** (which are a representation of collections HMIS projects) access to all data sets.  
```suggestion
```

Reply:
Thanks -  I'll take it out. I guess my thinking was that a data set is 'owned' by a data source, so having access to 'all data sources' would give you access to every data set (there is similar logic in policy provider related to projects and clients). But maybe the relationship between data set and data source isn't really ownership so this doesn't make sense.

But it sounds like it would be correct include the system data set group here:
`collection_ids += [Collection.system_collection(:supplemental_data_sets)&.id].compact`

Reply:
To find all collections that might include that data set, yes, the system collection for `:supplemental_data_sets` should include it, but won't that collection be included in `gve_scope.distinct.pluck(:collection_id)`?

Reply:
oh right! 🤦 I'll take it out